cranelift-codegen/src/isa/x86/abi.rs

```diff
@@ -115,9 +115,18 @@ impl ArgAssigner for Args {
         }
 
         // Try to use an FPR.
-        if ty.is_float() && self.fpr_used < self.fpr_limit {
-            let reg = FPR.unit(self.fpr_used);
-            self.fpr_used += 1;
+        let fpr_offset = if self.call_conv == CallConv::WindowsFastcall {
+            // For registers and Win64, the registers always depend on the parameter index.
+            // The first argument either uses RCX (int)/XMM0 (float) and the second RDX/XMM1.
+            // XMM0 is never used for the second parameter, even if the first parameter uses RCX
+            // and XMM0 theoretically would still be "available".
```

Comment:
This is nicely exhaustive. As a light suggestion, we could make it shorter by telling "Float and general registers on Win64 share the same parameter index, independently from the fact they're actually used or not" (of course, since I'm no native speaker, please fix grammar issues / reformulate to make it sound better etc. if you address this :))